src/server/storage/storageAPI.js

```diff
@@ -437,59 +438,84 @@ const setup = (app: Router, storage: StorageAPI) => {
         { mobile: mobile && sha3(mobile) }
-      ].filter(or => !!Object.values(or)[0])
+      ].filter(or => !!first(values(or)))
 
       if (queryOrs.length === 0) {
         log.warn('empty data for /userExists', { body: req.body })
-        return res.json({ ok: 0, exists: false })
+        sendNotExists()
+
+        return
       }
 
       let existing = await storage.model
         .find(
           {
             $or: queryOrs
           },
-          { identifier: 1, email: 1, mobile: 1, createdDate: 1, torusProvider: 1, fullName: 1 }
+          { identifier: 1, email: 1, mobile: 1, createdDate: 1, torusProvider: 1, fullName: 1, regMethod: 1, crmId: 1 }
         ) // sort by importance, prefer oldest verified account
         .sort({ isVerified: -1, createdDate: 1 })
         .lean()
 
       existing = existing.filter(doc => doc.createdDate)
 
-      if (identifierLC && (email || mobile)) {
+      if (lowerCaseID && (email || mobile)) {
         // if email or phone also were specified we want
         // to select matches by id first
         // sortBy sorts in ascending order (and keeps existing sort)
         // so non-matched by id results would be moved to the end
-        existing = sortBy(existing, ({ identifier }) => identifier !== identifierLC)
+        existing = sortBy(existing, ({ identifier }) => identifier !== lowerCaseID)
+      }
+
+      log.debug('userExists:', { existing, identifier, identifierLC: lowerCaseID, email, mobile })
+
+      if (!existing.length) {
+        sendNotExists()
+        return
       }
 
-      log.debug('userExists:', { existing, identifier, identifierLC, email, mobile })
+      const bestExisting = first(existing)
+      const foundByEmail = email && sha3(email) === bestExisting.email
 
-      if (existing.length) {
-        const bestExisting = first(existing)
+      // Won't fix passwordless/FB mobile logins, need to discuss separate solution for that
+      if (foundByEmail && !bestExisting.crmId) {
+        const { __utmzz: utmString = '' } = cookies
+        const crmUserRecord = { ...omit(bestExisting, 'mobile'), email }
 
-        return res.json({
-          ok: 1,
-          found: existing.length,
-          exists: true,
-          provider: bestExisting.torusProvider,
-          identifier: identifierLC === bestExisting.identifier,
-          email: email && sha3(email) === bestExisting.email,
-          mobile: mobile && sha3(mobile) === bestExisting.mobile,
-          fullName: bestExisting.fullName
-        })
+        log.warn('userExists:', 'Existing account have no crmId', { email })
+
+        if (mobile) {
+          assign(crmUserRecord, { mobile })
+        }
+
+        // fire and forget, don't wait for success or failure
+        createCRMRecord(crmUserRecord, utmString, log)
+          .then(() => log.debug('userExists: createCRMRecord success'))
+          .catch(e => log.error('userExists: createCRMRecord failed', e.message, e, { crmUserRecord }))
```

Comment:
i dont think this should be in userExists.
anyways the user is going to open a new account and the crm record will be created then
the fix for missing crm records should be done using the `updates` fix that we have